update-testsuite.sh

```diff
@@ -3,22 +3,125 @@ set -e
 set -u
 set -o pipefail
 
-spec="../spec"
+repos='spec threads simd exception-handling gc bulk-memory-operations tail-call nontrapping-float-to-int-conversions multi-value host-bindings sign-extension-ops mutable-global'
 
-pushd "$spec" >/dev/null
-    git checkout master
-    git fetch origin
-    git merge origin/master
-    id=$(git log --max-count=1 --pretty=oneline | sed -e 's/ .*//')
-popd >/dev/null
+log_and_run() {
+    echo ">>" $*
+    if ! $*; then
+        echo "sub-command failed: $*"
+        exit
+    fi
+}
 
-cp "$spec"/test/core/*.wast .
+try_log_and_run() {
+    echo ">>" $*
+    $*
+}
 
-for new in *.wast; do
-    git add "$new" || :
+pushdir() {
+    pushd $1 >/dev/null || exit
```

Comment:
The script uses "set -e", so it should automatically exit if pushd (or popd, below) fails.

Reply:
That's what I thought too, but it didn't seem to be failing when I tested it out... ah OK, so [this](https://stackoverflow.com/a/4073372) seems to be why. I guess the `set -e` was getting turned off when executed in the context of an `if`... maybe? weird stuff.

Reply:
Weird indeed.